src/calc/Calculator.java

```diff
@@ -156,13 +172,26 @@ Assoc getAssociativity(String op) {
         }
     }
 
+    public LinkedList<String> tokenize(String stringToTokenize) {
+        LinkedList<String> tokens = new LinkedList<>();
+        StringTokenizer stringTokenizer = new StringTokenizer(stringToTokenize);
+
+        while (stringTokenizer.hasMoreElements()) {
+            String token = stringTokenizer.nextToken();
+
+            // ??? ????? ????? ?? ?? ???
+            // maybe input from terminal is borken
+            if ("(".contains(token)) {
```

Comment:
Detta fattade jag inte riktigt. Jag dumpade även `token` som hex och jämförde med input från min terminal. De verkar vara identiska.

Egentligen skall endast `tokens.add(token);` behövas men det resulterar i att `token` senare tolkas som en operator i `getPrecedence()`.

Därför läggs paranteser in manuellt.